hermes_core/tests/test_util_util.py

```diff
@@ -234,3 +162,69 @@ def test_parse_science_filename_errors():
         # wrong instrument name
         f = "hermes_www_2s_l3test_burst_20240406_120621_v2.4.5"
         util.parse_science_filename(f)
+
+
+good_time = "2025-06-02T12:04:01"
+good_instrument = "eea"
+good_level = "l1"
+good_version = "1.3.4"
+# fmt: off
+@pytest.mark.parametrize("instrument,time,level,version", [
+    (good_instrument, good_time, good_level, "1.3"), # bad version specifications
+    (good_instrument, good_time, good_level, "1"),
+    (good_instrument, good_time, good_level, "1.5.6.7"),
+    (good_instrument, good_time, good_level, "1.."),
+    (good_instrument, good_time, good_level, "a.5.6"),
+    (good_instrument, good_time, "la", good_version),  # wrong level specifications
+    (good_instrument, good_time, "squirrel", good_version),
+    (good_instrument, good_time, "l0", good_version),
+    ("potato", good_time, good_level, good_version),  # wrong instrument names
+    ("eeb", good_time, good_level, good_version),
+    ("fpi", good_time, good_level, good_version),
+    (good_instrument, "2023-13-04T12:06:21", good_level, good_version),  # non-existent time
+    (good_instrument, "2023/13/04 12:06:21", good_level, good_version),  # not isot format
+    (good_instrument, "2023/13/04 12:06:21", good_level, good_version),  # not isot format
+    (good_instrument, "12345345", good_level, good_version),  # not valid input for time
+    ]
+)
+def test_science_filename_errors_l1_a(instrument, time, level, version):
+    """"""
+    with pytest.raises(ValueError) as e:
+        util.create_science_filename(
+            instrument, time, level=level, version=version
+        )
+# fmt: on
+
+
+def test_science_filename_errors_l1_b():
+    with pytest.raises(ValueError):
+        # _ character in mode
+        util.create_science_filename(
+            "eeb", time="12345345", level=good_level, version=good_version, mode="o_o"
+        )
+    with pytest.raises(ValueError):
+        # _ character in descriptor
+        util.create_science_filename(
+            "eeb",
+            time="12345345",
+            level=good_level,
+            version=good_version,
+            descriptor="blue_green",
+        )
+
+
+# fmt: off
+@pytest.mark.parametrize("filename,instrument,time,level,version", [
+    ("hermes_MAG_l0_2024094-124603_v01.bin", "nemisis", "2024-04-03T12:46:03", "l0", "01"),
+    ("hermes_EEA_l0_2026337-124603_v11.bin", "eea", "2026-12-03T12:46:03", "l0", "11"),
+    ("hermes_MRT_l0_2026215-124603_v21.bin", "merit", "2026-08-03T12:46:03", "l0", "21"),
```

Comment:
```suggestion
    ("hermes_MERIT_l0_2026215-124603_v21.bin", "merit", "2026-08-03T12:46:03", "l0", "21"),
```
This would also need to be changed to match the correct pneumonic.